File: frontend/lib/models/community.dart
class Community {
  const Community({
    required this.id,
    required this.name,
    required this.description,
    this.bannerUrl,
    this.postCount = 0,
    this.memberCount = 0,
  });

  final String id;
  final String name;
  final String description;
  final String? bannerUrl;
  final int postCount;
  final int memberCount;

  factory Community.fromJson(Map<String, dynamic> json) {
    return Community(
      id: json['id'].toString(),
      name: json['name']?.toString() ?? '',
      description: json['description']?.toString() ?? '',
      bannerUrl: json['image_url']?.toString(),
      postCount: (json['post_count'] as num?)?.toInt() ?? 0,
      memberCount: (json['member_count'] as num?)?.toInt() ?? 0,
    );
  }
}

File: backend/app/schemas/community.py
from typing import List, Optional
from pydantic import BaseModel


class Neighborhood(BaseModel):
    id: str
    name: str
    description: str
    image_url: Optional[str] = None
    post_count: int = 0
    member_count: int = 0
    is_member: bool = False


class NeighborhoodResponseModel(BaseModel):
    neighborhoods: List[Neighborhood]


class CommunityCreateRequest(BaseModel):
    name: str
    description: str
    latitude: float
    longitude: float
    # image: ?? Need help with this


class CommunityCreateResponse(BaseModel):
    community: Neighborhood
    message: str
